### ISSUE TITLE
Custom field can't be set. 

### ISSUE THREAD

**Issue by disco-lance** [tenable/integration-jira-cloud#109]
While trying to run the program for the first time, I'm getting the error below in the screenshot. We've verified all the information in the config.yaml is correct and the Jira account we're using is an admin. Any ideas?

![image](https://user-images.githubusercontent.com/81775531/113345213-c5521380-92f7-11eb-951c-543996b86a1e.png)

**Comment by SteveMcGrath** [comment]
can you please provide a run with the `--troubleshoot` flag set?

**Comment by nixdat** [comment]
Any update on this, as I'm experiencing the same issue 

be set. It is not on the appropriate screen, or unknown.","customfield_10230":"Field \'customfield_10230\' cannot be set. It is not on the appropriate screen, or unknown.","customfield_10110":"Field \'customfield_10110\' cannot be set. It is not on the appropriate screen, or unknown.","customfield_10231":"Field \'customfield_10231\' cannot be set. It is not on the appropriate screen, or unknown.","customfield_10232":"Field \'customfield_10232\' cannot be set. It is not on the appropriate screen, or unknown.","customfield_10113":"Field \'customfield_10113\' cannot be set. It is not on the appropriate screen, or unknown."}}'

**Comment by nixdat** [comment]
Just to update this thread, seems this issue is related to fields not associating correctly to the screens. Following this suggestion [https://github.com/tenable/integration-jira-cloud/issues/23#issuecomment-604631753](url) helped me resolve the issues

**Comment by SteveMcGrath** [comment]
@nixdat generally that's the solution.  @disco-lance never responded with the troubleshooting output, so we have simply been in a wait state for this issue.

The reasoning that happens is that Jira doesn't have any way to discern the linkage between the screens, the projects, and the issue types.  The integration attempts to figure it out, however it isn't always correct, and then you see these kinds of issues occur.  Id gets exacerbated as well when you have projects that create their own issue types and name them the same identical things as the global ones.

This integration tries to make these things easy, and it figures everything out correctly 90-95% of the time, but when it doesn't its a bit of a fishing expedition to suss it out.

I'm glad you were able to resolve the issue.

**Comment by fred-b-a9** [comment]
Curiously I'm facing the same issue now (--setup-only or regular run).
Have tried playing with the `screen/jira_ids`, `issue_types` pairs, no luck.
Version is latest (🤞🏼 1.1.23), python deps up to date. Jira user is full Jira admin.
 Attaching the latest debug log.
[tenable-setup-log_2021-04-27_1525PT.log](https://github.com/tenable/integration-jira-cloud/files/6388007/tenable-setup-log_2021-04-27_1525PT.log)
Some help would be greatly appreciated.

**Comment by SteveMcGrath** [comment]
@fred-b-a9 Have you tested all of the available issue types?  It looks like you have 3 separate sets of task and subtask issue types:

```yaml
11: Sub-task
3: Task
11210: Subtask
11208: Task
11220: Subtask
11218: Task
```

**Comment by fred-b-a9** [comment]
Yes @SteveMcGrath , I did try all the combinations possible between task ID ans sub-task ID, with and without specifying the screen / jira_ids. 
I re-read the issue #23 and ran the small python script to retrieve list of all issue types. Nothing other than the ones listed by the integration pops up.

**Comment by SteveMcGrath** [comment]
the screens seem to be properly linking in, hence all of the field logs when it validates that the screens are setup correctly.

I just went poking about in the UI in Jira, and when you click on a given issue type in the project settings, the issue type id is actually relayed back to you in the URL:

```
https://XXX.atlassian.net/plugins/servlet/project-config/VULN/issuetypes/10101/workflow
```

Note that in that case, the issuetype id is 10101 and is located here in the URL pathway: _/issuetypes/**10101**/workflow_

Take a look at each issuetype in the UI, add those IDs in, and lets see if that solves the issue.

**Comment by fred-b-a9** [comment]
IssueType IDs are 3 (task) and 11 (Sub-task). So, apparently they are properly detected 🤔 .
I'll raise a ticket on Atlassian side to see if they can help us troubleshooting this.

**Comment by SteveMcGrath** [comment]
it may be a permissions issue if that's the case.  Sadly Jira is very opaque in their errors when there is an issue.  Ensure that the user your using has ALL of the permissions detailed.  Yes I know in most cases the admin user _should_ work, but I've seen it go sideways in the past.

**Comment by fred-b-a9** [comment]
Some update, after getting feedback on our Jira Support Ticket.
It appears we have 3x pairs of Issue screens now with the key VUN (pairs because one is `Create` and the other `Edit/View`). Same name for all except suffix `(1)` or `(2)` for the extra pairs.
Jira tech told us that the screens referenced by the integration do not in fact have the fields added to them.
The `extra` screen IDs are not detected by the integration, apparently. 
I've updated the config file with those new IDs and ran the integration, it failed again but surprisingly the resulting log and config file show that IDs are the default detected ones, no more the custom ones I did set. 🤔

**Comment by SteveMcGrath** [comment]
Well from the last debug, it appears that the integration is assuming that the screens are 11460 and 11461.  If those are not the case (and you have multiple screens named the same thing) then you may want to hard-code the screen IDs.

If you look at the logs, the log message has 2 integer values separated by a colon (:).  That is the ScreenId:TabId that the fields have been attached to.

```
2021-04-27 15:26:27,701 tenable_jira.transform.Tio2Jira INFO Tenable Plugin Name already exists in 11461:11569
```

If you have multiple instances of the screens, I would suggest using the highest integer suffixes and see if that works.

**Comment by fred-b-a9** [comment]
I had tried that previously with no luck. 
What was also weird is that even after hardcoding specific issue task/sub-task ID, running the integration was overwriting them and reverting to the default ones (3 and 11).
Based on the feedback from Atlassian support, I have deleted all duplicate screens/schemes/... and kept the "original" ones, with ID 11460 and 11461.
After doing that, running the integration with screen's jira_ids also hardcoded seems to work now.
For some reason (probably the Jira API limitations you talked about in another issue), there is confusion if multiple screens/schemes have same name.
I do not understand why we had those duplicates initially 🤷🏼 .

Anyway, it seems solved for now 🤞🏼 . Thanks for helping.
